src/Schema/Traits/SortTrait.php

```
@@ -44,6 +45,12 @@ private static function getSortOrder(ResolveInfo $info, string $fieldName)
                     continue;
                 }
 
+                // If the sort has been passed as a variable, we can't attempt to fix it
+                // See https://github.com/silverstripe/silverstripe-graphql/issues/573
+                if (!$arg->value instanceof ObjectValueNode) {
+                    continue;
+                }
```

What are the circumstances that result in the value not being an instance of that class?
Depending on what those circumstances are, we might want to throw an exception (or at the very least log a warning) if someone tries to perform a sort that we explicitly don't respect here.

---

`$arg->value` is only ever an instance of `ObjectValueNode` when the sort order is baked into the query, so the `continue` statement is reached when sort is passed as a variable instead.